pkg/util/tencent/tencent.go

```diff
@@ -0,0 +1,101 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package tencent
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// declare these as vars not const to ease testing
+var (
+	metadataURL = "http://169.254.0.23"
+	timeout     = 300 * time.Millisecond
+
+	// CloudProviderName contains the inventory name of for CVM
+	CloudProviderName = "Tencent"
+)
+
+// IsRunningOn returns true if the agent is running on Tencent Cloud
+func IsRunningOn() bool {
+	if _, err := GetHostname(); err == nil {
+		return true
+	}
+	return false
+}
+
+// GetHostname fetches the hostname for current host from the Tencent metadata API
+func GetHostname() (string, error) {
+	return getMetadataItemWithMaxLength(metadataURL+"/meta-data/instance-name", config.Datadog.GetInt("metadata_endpoints_max_hostname_size"))
+}
```

Comment:
It seems we're using `GetHostname` only to detect if the API is there (from `IsRunningOn` and `diagnose`). Why not use `GetInstanceID` and rmeove `GetHostname` ?